src/software/thunderscope/gl/layers/gl_draw_polygon_obstacle.py

```diff
@@ -0,0 +1,148 @@
+from software.thunderscope.binary_context_managers.full_system import ProtoUnixIO
+from software.thunderscope.gl.graphics.gl_polygon import GLPolygon
+from pyqtgraph.Qt import QtGui
+from pyqtgraph.Qt.QtCore import Qt
+from pyqtgraph.opengl import *
+
+from proto.import_all_protos import *
+from software.py_constants import *
+from software.thunderscope.gl.helpers.observable_list import ObservableList
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+
+from software.thunderscope.gl.layers.gl_layer import GLLayer
+from software.thunderscope.gl.helpers.extended_gl_view_widget import MouseInSceneEvent
+
+
+class GLDrawPolygonObstacleLayer(GLLayer):
+    """A layer used to draw polygons that are going to represent obstacles for the trajectroy planner
+    to avoid.
+    """
+
+    def __init__(
+        self, name, blue_fs_io: ProtoUnixIO, yellow_fs_io: ProtoUnixIO
```

Comment:
we only need to send protos to one fullsystem (the full system we are running the robots with)